ios/src/CGFontConverter.mm

```diff
@@ -0,0 +1,154 @@
+//
+//  CGFontConverter.m
+//  tangram
+//
+//  Created by Karim Naaji on 11/01/16.
+//
+//
+
+#import "CGFontConverter.h"
+
+#include <cstdio>
+
+struct FontHeader {
+    int32_t fVersion;
+    uint16_t fNumTables;
+    uint16_t fSearchRange;
+    uint16_t fEntrySelector;
+    uint16_t fRangeShift;
+};
+
+typedef struct FontHeader FontHeader;
+
+struct TableEntry {
+    uint32_t fTag;
+    uint32_t fCheckSum;
+    uint32_t fOffset;
+    uint32_t fLength;
+};
+
+typedef struct TableEntry TableEntry;
+
+static uint32_t CalcTableCheckSum(const uint32_t *table, uint32_t numberOfBytesInTable)
+{
+    uint32_t sum = 0;
+    uint32_t nLongs = (numberOfBytesInTable + 3) / 4;
+    while (nLongs-- > 0) {
+       sum += CFSwapInt32HostToBig(*table++);
+    }
+    return sum;
+}
+
+static uint32_t CalcTableDataRefCheckSum(CFDataRef dataRef)
+{
+    const uint32_t* dataBuff = (const uint32_t *)CFDataGetBytePtr(dataRef);
+    uint32_t dataLength = (uint32_t)CFDataGetLength(dataRef);
+    return CalcTableCheckSum(dataBuff, dataLength);
+}
+
+@implementation CGFontConverter
+
+//Reference:
+//http://skia.googlecode.com/svn-history/r1473/trunk/src/ports/SkFontHost_mac_coretext.cpp
+//https://gist.github.com/Jyczeal/1892760
+
++ (unsigned char *)fontDataForCGFont:(CGFontRef)cgFont size:(size_t *)size
+{
+    if (!cgFont) {
+        return nil;
+    }
+
+    CFRetain(cgFont);
+
+    CFArrayRef tags = CGFontCopyTableTags(cgFont);
+    int tableCount = CFArrayGetCount(tags);
+
+    size_t* tableSizes = (size_t*)malloc(sizeof(size_t) * tableCount);
+    memset(tableSizes, 0, sizeof(size_t) * tableCount);
+
+    BOOL containsCFFTable = NO;
+
+    size_t totalSize = sizeof(FontHeader) + sizeof(TableEntry) * tableCount;
+
+    for (int index = 0; index < tableCount; ++index) {
+
```

Comment:
I'm concerned about the performance implications about iterating the same iterator set twice. Any way to combine this loop and the one at L128?